src/libmongoc/src/mongoc/mongoc-uri.h

```diff
@@ -197,6 +198,10 @@ MONGOC_EXPORT (const mongoc_read_concern_t *)
 mongoc_uri_get_read_concern (const mongoc_uri_t *uri);
 MONGOC_EXPORT (void)
 mongoc_uri_set_read_concern (mongoc_uri_t *uri, const mongoc_read_concern_t *rc);
+MONGOC_EXPORT (const char *)
+mongoc_uri_get_server_monitoring_mode (const mongoc_uri_t *uri);
```

Comment:
Add .rst documentation for the new public API (example: [mongoc_uri_set_auth_mechanism](https://mongoc.org/libmongoc/current/mongoc_uri_set_auth_mechanism.html)).

Aside: some URI functions are unfortunately missing public documentation (e.g. `mongoc_uri_set_appname`, noted in CDRIVER-3473).